src/Expander/Monad.hs

```diff
@@ -365,10 +367,20 @@ linkPattern :: PatternPtr -> ConstructorPattern -> Expand ()
 linkPattern dest pat =
   modifyState $ over expanderCompletedPatterns (<> Map.singleton dest pat)
 
-linkDecl :: DeclPtr -> Decl SplitTypePtr SchemePtr DeclPtr SplitCorePtr -> Expand ()
+linkModBody :: ModBodyPtr -> ModuleBodyF DeclPtr ModBodyPtr -> Expand ()
+linkModBody dest moduleBodyF =
+  modifyState $ over expanderCompletedModBody $ (<> Map.singleton dest moduleBodyF)
+
+linkDecl :: DeclPtr -> Decl SplitTypePtr SchemePtr ModBodyPtr SplitCorePtr -> Expand ()
 linkDecl dest decl =
   modifyState $ over expanderCompletedDecls $ (<> Map.singleton dest decl)
 
+linkOneDecl :: ModBodyPtr -> Decl SplitTypePtr SchemePtr ModBodyPtr SplitCorePtr -> Expand ()
```

Comment:
The combination `Decl SplitTypePtr SchemePtr ModBodyPtr SplitCorePtr` occurs often enough that it should probably be given a name. Maybe `ShallowDecl`?

Reply:
What about `SplitDecl`, by analogy to the other spit things it wraps? Should we `newtype` it to make type error messages less complicated?

Reply:
Hmm, it does wrap `SplitTypePtr` and `SplitCorePtr`, but it wraps `DeclTreePtr` (formerly `ModBodyPtr`), not `SplitDeclTree`, and it wraps `SchemePtr` not `SplitScheme` (for the good reason that `SplitScheme` doesn't exist). Is this perhaps an indication that it _should_ wrap `SplitDeclTree` and `SplitScheme` instead?

Reply:
I think I'll just leave it alone.